src/main/java/com/Nearsoft/referrals/service/impl/GoogleTokenVerifyServiceImpl.java

```diff
@@ -0,0 +1,62 @@
+package com.Nearsoft.referrals.service.impl;
+
+import com.Nearsoft.referrals.service.GoogleTokenVerifyService;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.*;
+import java.security.GeneralSecurityException;
+import java.util.Arrays;
+
+@Service
+public class GoogleTokenVerifyServiceImpl implements GoogleTokenVerifyService {
+
+    private static final JacksonFactory jacksonFactory = new JacksonFactory();
+    private String iosClientId;
+    private String androidClientId;
+
+    public GoogleTokenVerifyServiceImpl(@Value("${ios.clientId}") String iosId,@Value("${android.clientId}") String androidClientId) {
+        this.iosClientId = iosId;
+        this.androidClientId = androidClientId;
+    }
+
+    @Override
+    public Boolean verifyToken(String idTokenString) throws GeneralSecurityException, IOException {
+        GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier.Builder(new NetHttpTransport(),jacksonFactory)
+            // Specify the CLIENT_ID of the apps that accesses the backend:
+            .setAudience(Arrays.asList(iosClientId, androidClientId))
+            .build();
+
+        // (Receive idTokenString by HTTPS POST)
+
+        GoogleIdToken idToken = verifier.verify(idTokenString);
+        if (idToken != null) {
```

Comment:
I recommend to use something like 
`if(idToken == null) {
    return false
}` 
Then continue without nest the code in any block.